lib/private/Log.php

```diff
@@ -281,6 +293,22 @@ public function getLogLevel($context): int {
 			}
 		}
 
+		if (isset($logCondition['matches'])) {
+			foreach ($logCondition['matches'] as $option) {
+				if (
+					(!isset($option['shared_secret']) || $this->checkLogSecret($option['shared_secret']))
+					&& (!isset($option['users']) || in_array($userId, $option['users'], true))
+					&& (!isset($option['apps']) || (isset($context['app']) && in_array($context['app'], $option['apps'], true)))
+					&& (!isset($option['message']) || str_contains($message, $option['message']))
+				) {
+					if (!isset($option['apps']) && !isset($option['loglevel']) && !isset($option['message'])) {
+						$this->logConditionSatisfied = true;
+					}
+					return $option['loglevel'] ?? ILogger::DEBUG;
+				}
+			}
```

Comment:
Run through the function recursively maybe?
```suggestion
				$loglevel = $this->getLogLevel($option, $option['message'] ?? $message);
			}
```

Reply:
I didn't understand this one.
Fixed all the other comments